src/Core/Ripgrep.re

```diff
@@ -14,7 +32,60 @@ let getCompletedCount = () => _ripGrepCompletedCount^;
 type searchFunction =
   (string, list(string) => unit, unit => unit) => disposeFunction;
 
-type t = {search: searchFunction};
+module Match = {
+  type t = {
+    file: string,
+    text: string,
+    lineNumber: int,
+    charStart: int,
+    charEnd: int,
+  };
+
+  let fromJsonString = str => {
+    Yojson.Basic.Util.(
```

Comment:
Just curious - when do you prefer `Yojson.Basic` to `Yojson.Safe` - I'm not very aware of the differences?

Reply:
I wasn't even aware of it until you mentioned it just now. Or rather, I think I've seen it mentioned before, but had no awareness of the difference between that and anything else. But it seems like `Yojson.Safe` is a superset of `Yojson.Basic` with some "extended" and non-standard primitives. How that makes it "safe" I don't know, but it seems to offer some more convenience if both producer and consumer is using OCaml, and `Yojson.Safe` in particular.